src/Compilers/CSharp/Portable/BoundTree/Statement.cs

```diff
@@ -445,7 +445,14 @@ internal partial class BoundCatchBlock : ICatchClause
 
         IOperation ICatchClause.Filter => this.ExceptionFilterOpt;
 
-        ILocalSymbol ICatchClause.ExceptionLocal => this.LocalOpt;
+        ILocalSymbol ICatchClause.ExceptionLocal
+        {
+            get
+            {
+                var local = this.Locals.FirstOrDefault();
+                return local?.DeclarationKind == LocalDeclarationKind.CatchVariable ? local : null;
```

Comment:
A comment in the usage couldn't hurt either -- comments in the xml file are easy to miss if you only read the generated code (even if that's not a good idea).